mesonbuild/options.py

```diff
@@ -1101,6 +1101,14 @@ def get_option_from_meson_file(self, key: OptionKey) -> 'T.Tuple[AnyOptionType,
         (value_object, value) = self.get_value_object_and_value_for(key)
         return (value_object, value)
 
+    def get_default_for_b_option(self, keyname: str) -> 'T.Tuple[AnyOptionType, OptionValueType]':
+        assert keyname.startswith('b_')
+        from .compilers.compilers import BASE_OPTIONS
+        for bkey, bvalue in BASE_OPTIONS.items():
+            if bkey.name == keyname:
+                return (bvalue, bvalue.default) # type: ignore [return-value]
```

Comment:
What is wrong with the return type?

Reply:
They are the same, but in `compilers.py` it is written out as a union rather than being an `AnyOptionType`.

Reply:
default is annotated as an `Any` in compilers.py (somewhere I have a patch that fixes that...)

So I think this should be `return (bvalue, T.cast('options.AnyOptionType', bvalue.default))`.